core/builder/src/main/java/io/quarkus/builder/Consume.java

```diff
@@ -1,44 +1,20 @@
 package io.quarkus.builder;
 
-/**
- */
-final class Consume {
-    private final BuildStepBuilder buildStepBuilder;
-    private final ItemId itemId;
-    private final Constraint constraint;
-    private final ConsumeFlags flags;
+import static io.quarkus.builder.Constraint.ORDER_ONLY;
+import static io.quarkus.builder.Constraint.REAL;
+import static io.quarkus.builder.ConsumeFlag.OPTIONAL;
 
-    Consume(final BuildStepBuilder buildStepBuilder, final ItemId itemId, final Constraint constraint,
-            final ConsumeFlags flags) {
-        this.buildStepBuilder = buildStepBuilder;
-        this.itemId = itemId;
-        this.constraint = constraint;
-        this.flags = flags;
-    }
-
-    BuildStepBuilder getBuildStepBuilder() {
-        return buildStepBuilder;
-    }
-
-    ItemId getItemId() {
-        return itemId;
-    }
-
-    ConsumeFlags getFlags() {
-        return flags;
-    }
-
-    Consume combine(final Constraint constraint, final ConsumeFlags flags) {
-        final Constraint outputConstraint = constraint == Constraint.REAL || this.constraint == Constraint.REAL
-                ? Constraint.REAL
-                : Constraint.ORDER_ONLY;
-        final ConsumeFlags outputFlags = !flags.contains(ConsumeFlag.OPTIONAL) || !this.flags.contains(ConsumeFlag.OPTIONAL)
-                ? flags.with(this.flags).without(ConsumeFlag.OPTIONAL)
-                : flags.with(this.flags);
-        return new Consume(buildStepBuilder, itemId, outputConstraint, outputFlags);
-    }
+public record Consume(BuildStepBuilder buildStepBuilder, ItemId itemId, Constraint constraint, ConsumeFlags flags) {
```

Comment:
Hey,

why is package-private visibility turned into public visibility? Is this class supposed to be used outside of this package with your changes?

Reply:
That's the reason I didn't merge this yet. I wanted to look at where this is actually used on my machine instead of my phone

Reply:
indeed, good finding even twice. thanks